temporal-opentracing/src/main/java/io/temporal/opentracing/internal/OpenTracingContextAccessor.java

```diff
@@ -0,0 +1,95 @@
+/*
+ *  Copyright (C) 2020 Temporal Technologies, Inc. All Rights Reserved.
+ *
+ *  Copyright 2012-2016 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ *  Modifications copyright (C) 2017 Uber Technologies, Inc.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License"). You may not
+ *  use this file except in compliance with the License. A copy of the License is
+ *  located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ *  or in the "license" file accompanying this file. This file is distributed on
+ *  an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *  express or implied. See the License for the specific language governing
+ *  permissions and limitations under the License.
+ */
+
+package io.temporal.opentracing.internal;
+
+import com.google.gson.reflect.TypeToken;
+import io.opentracing.SpanContext;
+import io.opentracing.Tracer;
+import io.opentracing.propagation.Format;
+import io.opentracing.propagation.TextMap;
+import io.temporal.api.common.v1.Payload;
+import io.temporal.common.converter.DataConverter;
+import io.temporal.common.interceptors.Header;
+import java.lang.reflect.Type;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.Map;
+import java.util.Optional;
+
+public class OpenTracingContextAccessor {
+  private static final String TRACER_HEADER_KEY = "_tracer-data";
+  private static final Type HASH_MAP_STRING_STRING_TYPE =
+      TypeToken.getParameterized(HashMap.class, String.class, String.class).getType();
+
+  public static void writeSpanContextToHeader(
+      SpanContext spanContext, Header header, Tracer tracer) {
+    Map<String, String> serializedSpanContext = serializeSpanContextToMap(spanContext, tracer);
+    Optional<Payload> payload = DataConverter.getDefaultInstance().toPayload(serializedSpanContext);
+    header.getValues().put(TRACER_HEADER_KEY, payload.get());
+  }
+
+  private static Map<String, String> serializeSpanContextToMap(
+      SpanContext spanContext, Tracer tracer) {
+    HashMapBackedTextMap contextTextMap = new HashMapBackedTextMap();
+    tracer.inject(spanContext, Format.Builtin.TEXT_MAP, contextTextMap);
+    return contextTextMap.getBackingMap();
+  }
+
+  public static SpanContext readSpanContextFromHeader(Header header, Tracer tracer) {
+    Payload payload = header.getValues().get(TRACER_HEADER_KEY);
+    if (payload == null) {
+      return null;
+    }
+    Map<String, String> serializedSpanContext =
+        DataConverter.getDefaultInstance()
+            .fromPayload(payload, HashMap.class, HASH_MAP_STRING_STRING_TYPE);
+    return deserializeSpanContextFromMap(serializedSpanContext, tracer);
+  }
+
+  private static SpanContext deserializeSpanContextFromMap(
+      Map<String, String> serializedSpanContext, Tracer tracer) {
+    HashMapBackedTextMap contextTextMap = new HashMapBackedTextMap(serializedSpanContext);
+    return tracer.extract(Format.Builtin.TEXT_MAP, contextTextMap);
```

Comment:
you can use io.opentracing.propagation.TextMapAdapter from opentracing-api.jar and remove local HashMapBackedTextMap.
```java
return tracer.extract(Format.Builtin.TEXT_MAP, new TextMapAdapter(serializedSpanContext));
```

Reply:
That's neat, I replaced this custom `HashMapBackedTextMap` class with standard opentracing classes from `io.opentracing.propagation` package. Thank you for pointing this out!